tests/test_bundle_run.py

```diff
@@ -0,0 +1,136 @@
+        },
+        "dataloader": {
+            "<name>": "DataLoader",
+            "<args>": {"dataset": "@dataset", "batch_size": 1, "shuffle": False, "num_workers": 4},
+        },
+        "inferer": {
+            "<name>": "SlidingWindowInferer",
+            "<args>": {"roi_size": [96, 96, 96], "sw_batch_size": 4, "overlap": 0.5},
+        },
+        "postprocessing": {
+            "<name>": "Compose",
+            "<args>": {
+                "transforms": [
+                    {"<name>": "Activationsd", "<args>": {"keys": "pred", "softmax": True}},
+                    {"<name>": "AsDiscreted", "<args>": {"keys": "pred", "argmax": True}},
+                    {
+                        "<name>": "SaveImaged",
+                        "<args>": {
+                            "keys": "pred",
+                            "meta_keys": "image_meta_dict",
+                            "output_dir": "@<meta>#output_dir",  # test placeholger with `output_dir`
+                        },
+                    },
+                ]
+            },
+        },
+        "evaluator": {
+            "<name>": "SupervisedEvaluator",
+            "<args>": {
+                "device": "@device",
+                "val_data_loader": "@dataloader",
+                "network": "@network",
+                "inferer": "@inferer",
+                "postprocessing": "@postprocessing",
+                "amp": False,
+            },
+        },
+    },
+    (128, 128, 128),
+]
+
+
+class TestBundleRun(unittest.TestCase):
+    @parameterized.expand([TEST_CASE_1])
+    def test_shape(self, config, expected_shape):
+        test_image = np.random.rand(128, 128, 128)
+        with tempfile.TemporaryDirectory() as tempdir:
+            filename = os.path.join(tempdir, "image.nii")
+            nib.save(nib.Nifti1Image(test_image, np.eye(4)), filename)
+
+            meta = {"datalist": [{"image": filename}], "output_dir": tempdir, "window": (96, 96, 96)}
+            # test YAML file
+            metafile = os.path.join(tempdir, "meta.yaml")
+            with open(metafile, "w") as f:
+                yaml.dump(meta, f)
+            # test JSON file
+            configfile = os.path.join(tempdir, "config.json")
+            with open(configfile, "w") as f:
+                json.dump(config, f)
+
+            # test override with file, up case postfix
+            overridefile1 = os.path.join(tempdir, "override1.JSON")
+            with open(overridefile1, "w") as f:
+                # test override with part of the overriding file
+                json.dump({"move_net": "$@network_def.to(@device)"}, f)
+            overridefile2 = os.path.join(tempdir, "override2.JSON")
+            with open(overridefile2, "w") as f:
+                # test override with the whole overriding file
+                json.dump("Dataset", f)
+
+            ret = os.system(
```

Comment:
`os.system` isn't a reliable function to use because it depends on creating a new shell and the environment that has isn't well controlled. This will likely fail in certain Anaconda environments for example. It will also be sensitive to how the command is parsed which will vary by platform. I would recommend `subprocess.check_call` with `sys.executable` as the path to the interpreter itself so that you don't have to worry about what `python` is aliasing.

Reply:
Good point, updated in the latest commit.

Thanks.

Reply:
Hi @ericspod ,

When I changed to `subprocess`, the windows CI test can't pass:
https://github.com/Project-MONAI/MONAI/runs/5394963470?check_suite_focus=true#step:10:579
Maybe we need another command for windows platform?
```
======================================================================
ERROR: test_shape_0 (tests.test_bundle_run.TestBundleRun)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\site-packages\parameterized\parameterized.py", line 533, in standalone_func
    return func(*(a + p.args), **p.kwargs)
  File "D:\a\MONAI\MONAI\tests\test_bundle_run.py", line 134, in test_shape
    ret = subprocess.check_call(
  File "C:\hostedtoolcache\windows\Python\3.8.10\x64\lib\subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'C:\hostedtoolcache\windows\Python\3.8.10\x64\python.exe -m monai.bundle.scripts run --meta_file=C:\Users\RUNNER~1\AppData\Local\Temp\tmpnlqqp0ze\meta.yaml --config_file=C:\Users\RUNNER~1\AppData\Local\Temp\tmpnlqqp0ze\config.json --override='{postprocessing#<args>#transforms#2#<args>#output_postfix: seg, network: <file>C:\Users\RUNNER~1\AppData\Local\Temp\tmpnlqqp0ze\override1.JSON#move_net, dataset#<name>: <file>C:\Users\RUNNER~1\AppData\Local\Temp\tmpnlqqp0ze\override2.JSON}' --target=evaluator' returned non-zero exit status 1.
```

Thanks.

Reply:
I skipped the `subprocess` test for windows.

Thanks.